lib/task/import/importDipObjectsTask.class.php

```diff
@@ -0,0 +1,585 @@
+   * objects to.
+   *
+   * @param resource $fh  File pointer resource for CSV file
+   *
+   * @return void
+   */
+  protected function processCsvHeaderRow($fh)
+  {
+    // Storage order of columns in CSV header row
+    $this->columnNames = fgetcsv($fh, 1000);
+
+    // Make sure there isn't both an "identifier" and "slug" column in the CSV header row
+    if (in_array('identifier', $this->columnNames) && in_array('slug', $this->columnNames))
+    {
+      throw new sfException('Error: CSV header row includes both an "identifier" column and a "slug" column. Please use only one of the two.');
+    }
+
+    // Make sure there is either an "identifier" or a "slug" column  in the CSV header row
+    if (!in_array('identifier', $this->columnNames) && !in_array('slug', $this->columnNames))
+    {
+      throw new sfException('Error: CSV header row must include either an "identifier" column or a "slug" column.');
+    }
+
+    // Determine which column name should be used to specific individual information objects
+    $this->uniqueValueColumnName = (in_array('identifier', $this->columnNames)) ? 'identifier' : 'slug';
+  }
+
+  /**
+   * Get value of column, using column name, from array representing row of CSV data
+   *
+   * @return void
+   */
+  function getRowColumnValue($column, $row)
+  {
+    if (is_numeric($columnIndex = array_search($column, $this->columnNames)))
+    {
+      return $row[$columnIndex];
+    }
+
+    throw new sfException('Missing column "'. $column .'".');
+  }
+
+  /**
+   * Import digital objects
+   *
+   * @param array $digitalObjects  Hash with keys of unique value for
+   *                               specifying information objects, values:
+   *                               digital object filenames
+   * @param boolean $auditMode  Whether to audit, rather than import, objects
+   *
+   * @return void
+   */
+  protected function importDigitalObjects($digitalObjects, $auditMode = false)
+  {
+    $count = 0;
+
+    // Loop through $digitalObject hash and add digital objects to db
+    foreach ($digitalObjects as $key => $item)
+    {
+      $opDescription = ($auditMode) ? 'Auditing' : 'Importing to';
+      $this->logSection(sprintf("%s '$key'...", $opDescription));
+
+      if ($auditMode)
+      {
+        if (!is_array($item))
+        {
+          $this->auditDigitalObject($item);
+          $count++;
+        }
+        else
+        {
+          // If more than one digital object linked to this information object
+          $parent = $this->getInformationObjectUsingUniqueId($key);
+
+          for ($i = 0; $i < count($item); $i++)
+          {
+            $this->auditDigitalObject($item[$i]);
+            $count++;
+          }
+        }
+      }
+      else
+      {
+        $informationObject = $this->getInformationObjectUsingUniqueId($key);
+
+        if (!is_array($item))
+        {
+          $this->addDigitalObject($informationObject, $item, $undoLog);
```

Comment:
I think $undoLog is not here, you may need to make it global or send it to this function (importDigitalObjects).